### PR TITLE
Add WinRT API to spec for sharing security descriptors

### PR DESCRIPTION
**Pull request by DefaultRyan**
Add WinRT API to spec for sharing security descriptors

### CODE REVIEW

File: specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md [hunk 1]
```diff
@@ -61,6 +61,26 @@ STDAPI GetSecurityDescriptorForAppContainerNames(
...
+{
+  struct AppContainerAccess
+  {
+    String appContainerName;
+    UInt32 accessMask;
+  }
+
+  static runtimeclass SecurityDescriptorHelpers
+  {
+    String GetSDDLForAppContainerNames(
```
Comment (jonwis) [inline specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: Does this also need a 'static' here? Or is "class level static" sufficient?

Comment (DefaultRyan) [reply specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: Good callout. Apparently, I don't author enough MIDL3 to remember the rules here, but you're correct, I do need `static` on the method.

https://docs.microsoft.com/en-us/uwp/midl-3/intro#classes

**Review by mziller — COMMENTED** [review]
With the added WinRT APIs, the full doc here still reads as specific to Win32 callers and doesn't describe in much detail usage of the new GetSDDLForAppContainerNames method (e.g. an example use case like shown for Win32)

File: specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md [hunk 2]
```diff
@@ -61,6 +61,26 @@ STDAPI GetSecurityDescriptorForAppContainerNames(
 
 If the function succeds, the returned `SECURITY_DESCRIPTOR` must be freed by calling [LocalFree](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-localfree).
 
+## WinRT
+```c#
+namespace Microsoft.Security.AccessControl
+{
+  struct AppContainerAccess
```
Comment (ptorr-msft) [inline specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:68]: Update name to match native one, if necessary. 
Also, why struct? Win32 structs are extensible, but WinRT structs are not. Should this be a runtime class?

File: specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md [hunk 3]
```diff
@@ -61,6 +61,26 @@ STDAPI GetSecurityDescriptorForAppContainerNames(
...
+  {
+    String appContainerName;
+    UInt32 accessMask;
+  }
+
+  static runtimeclass SecurityDescriptorHelpers
+  {
+    static String GetSDDLForAppContainerNames(
```
Comment (ptorr-msft) [inline specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: Strange that this generates the SDDL but the native one generates the actual `SECURITY_DESCRIPTOR`. Is there any reasonable way you can use the SDDL in a projection where you couldn't use the SD? (In other words, does this WinRT API give you a result that effectively requires you to call into Win32 anyway?)

Comment (oldnewthing) [reply specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: You could use the SDDL in places like manifests. (There's really no good way to return a `SECURITY_DESCRIPTOR` to WinRT; maybe a `byte[]`?)

Comment (oldnewthing) [inline specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: Presumable `Sddl` per capitalization guidelines.

File: specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md (hunk 2 above)
Comment (jonwis) [reply specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:68]: RECOMMEND: Yes, rename to match the C++ one.

RECOMMEND: Switch to runtimeclass here for less breakage

File: specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md (hunk 3 above)
Comment (jonwis) [reply specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: CONSIDER: Alternate `byte[] GetSecurityDescriptorBytesFromAppContainerNames` for use with .NET's `RawSecurityDescriptor` constructor. Use customer feedback to know what's important.

Comment (jonwis) [reply specs/GetSecurityDescriptorForAppContainerNames/GetSecurityDescriptorForAppContainerNames.md:76]: RECOMMEND: Yes

**Comment by jonwis** [comment]
> With the added WinRT APIs, the full doc here still reads as specific to Win32 callers and doesn't describe in much detail usage of the new GetSDDLForAppContainerNames method (e.g. an example use case like shown for Win32)

RECOMMEND: Show a WinRT sample using the API and populating a RawSecurityDescriptor with it for .NET

**Review by jonwis — APPROVED** [review]

**Pull request merged**